### ISSUE TITLE
errors running ganeti jobs if /proc/sys/fs/inotify/max_user_watches is too low

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#274]
Originally reported of Google Code with ID 218.

```
What software version are you running? Please provide the output of "gnt-cluster --version" and "gnt-cluster version".
Software version: 2.4.5
Internode protocol: 2040000
Configuration format: 2040000
OS api version: 20
Export interface: 0

<b>What distribution are you using?</b>
Debian Squeeze 6.0

<b>What steps will reproduce the problem?</b>
1. lower inotify max_user_watches
   (echo 4000 > /proc/sys/fs/inotify/max_user_watches)
2. run gnt-cluster verify
(if no error appears, lower max_user_watches even more)

<b>What is the expected output? What do you see instead?</b>
Ganeti should warn about a too low max_user_watches
error reported instead:
Error checking job status: Job with id 6311 lost

Warning that inotifywatch /tmp reports if the number is too low:
Failed to watch /tmp/; upper limit on inotify watches reached!
Please increase the amount of inotify watches allowed per user via `/proc/sys/fs/inotify/max_user_watches'.

<b>Please provide any additional information below.</b>
I have Crashplan installed on the same machine as Ganeti. Crashplan uses inotify as well and already increased the default of 8192 to 16384 in /etc/sysctl.conf.
```
Originally added on 2012-02-24 13:32:15 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Can Crashplan exclude a specific directory? Excluding /var/lib/ganeti/queue would seem sensible. There's a lot of I/O with atomically replaced files going on.
```
Originally added on 2012-02-27 16:38:42 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Crashplan can exclude that directory, yes. But it doesn't help the fact that it will crash when max_user_watches is set too low.
```
Originally added on 2012-03-01 09:14:56 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Yes, we will look into improving error reporting and handling.
```
Originally added on 2012-03-01 11:06:32 +0000 UTC.

Changed State: Accepted

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-07-19 14:55:15 +0000 UTC.

Added Labels: Type-Defect

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-11-07 14:49:06 +0000 UTC.

Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 383477e
Author: Michael Hanselmann <hansmi@google.com>
Date:   Mon Dec 10 14:32:13 2012 +0100

    jqueue: Improve inotify error reporting
    
    This addresses issue 218. When the number of inotify watches is
    exhausted, for example by being set too low from the beginning or by
    other programs, waiting for a job to change would just report a lost job
    (e.g. “Error checking job status: Job with id 7817 lost”).
    
    This patch changes the job watcher to no longer catch
    “errors.InotifyError” and, this is by far the larger part of this patch,
    adds unittests for this situation.
    
    Signed-off-by: Michael Hanselmann <hansmi@google.com>
    Reviewed-by: Guido Trotter <ultrotter@google.com>

```
Originally added on 2012-12-13 11:49:45 +0000 UTC.

Mentioned Issue with GoogleCode ID 218 has been migrated to GitHub with ID #274.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:48 +0000 UTC.

Changed State: Released
